src/main/java/org/geysermc/discordbot/commands/FloodgateUuidCommand.java

```diff
@@ -0,0 +1,104 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/GeyserDiscordBot
+ */
+
+package org.geysermc.discordbot.commands;
+
+import com.jagrosh.jdautilities.command.SlashCommand;
+import com.jagrosh.jdautilities.command.SlashCommandEvent;
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.commands.build.OptionData;
+import org.geysermc.discordbot.util.BotColors;
+import org.json.JSONException;
+import org.json.JSONObject;
+import pw.chew.chewbotcca.util.RestClient;
+
+import java.util.Collections;
+import java.util.UUID;
+
+public class FloodgateUuidCommand extends SlashCommand {
+
+    public FloodgateUuidCommand() {
+        this.name = "uuid";
+        this.help = "Get the floodgate uuid from a bedrock username.";
+        this.arguments = "<bedrock-username>";
+        this.guildOnly = false;
+        this.cooldown = 60;
+
+        this.options = Collections.singletonList(
+                new OptionData(OptionType.STRING, "bedrock-username", "username to grab floodgate uuid from").setRequired(true)
+        );
+    }
+
+    @Override
+    protected void execute(SlashCommandEvent event) {
+        // get bedrock username, replace char in case they include Floodgate prefix.
+        String username = event.optString("bedrock-username","").replace(".", "");
+        try {
+            // get xuid as json object and convert xuid into Floodgate uuid
+            JSONObject xuid = new JSONObject(RestClient.get("https://api.geysermc.org/v2/xbox/xuid/" + username));
+            // if message gets returned = floodgate player wasn't found
+            if (xuid.has("message")) {
+                event.replyEmbeds(floodgateUUID(username, null, false,0)).queue();
+                return;
+            }
+
+            if (xuid.has("xuid")) {
+                UUID floodgateUUID = new UUID(0, xuid.getLong("xuid"));
+                event.replyEmbeds(floodgateUUID(username, floodgateUUID, false, xuid.getLong("xuid"))).queue();
+            }
+            // this occurs when api wasn't available
+        } catch (JSONException e) {
+            event.replyEmbeds(floodgateUUID(username, null, true,0)).queue();
```

Comment:
Why do we not just make and submit a failure embed here instead of putting it in the method?